pgjdbc/src/main/java/org/postgresql/util/PSQLState.java

```diff
@@ -118,12 +121,29 @@ public String getState() {
     return this.state;
   }
 
-  public static boolean isConnectionError(@Nullable String psqlState) {
-    return PSQLState.CONNECTION_UNABLE_TO_CONNECT.getState().equals(psqlState)
-        || PSQLState.CONNECTION_DOES_NOT_EXIST.getState().equals(psqlState)
-        || PSQLState.CONNECTION_REJECTED.getState().equals(psqlState)
-        || PSQLState.CONNECTION_FAILURE.getState().equals(psqlState)
-        || PSQLState.CONNECTION_FAILURE_DURING_TRANSACTION.getState().equals(psqlState);
+  private static Map<String, PSQLState> codeToEnum;
+
+  @Nullable
+  public static PSQLState fromCode(final String string) {
+    if (codeToEnum == null) {
+      final PSQLState[] values = PSQLState.values();
+      final Map<String, PSQLState> temp = new HashMap<>(values.length);
```

Comment:
This will not size the map large enough to avoid a resize.
Need to use Math.ceil((int) (values.length/0.75))